vendor/sigs.k8s.io/apiserver-network-proxy/konnectivity-client/pkg/client/client.go

```diff
@@ -102,13 +108,19 @@ func (t *grpcTunnel) serve(c *grpc.ClientConn) {
 			t.pendingDialLock.RUnlock()
 
 			if !ok {
-				klog.Warning("DialResp not recognized; dropped")
+				klog.V(1).Infoln("DialResp not recognized; dropped")
 			} else {
 				ch <- dialResult{
 					err:    resp.Error,
 					connid: resp.ConnectID,
 				}
 			}
+
+			if resp.Error != "" {
+				// On dial error, avoid leaking serve goroutine.
+				return
```

Comment:
@liggitt can you help approve this cherrypick? Thanks.

This line fixes a memory leak. Other changes are just migrating to structured logging.